code/game/objects/items/cigs_lighters.dm

```diff
@@ -322,9 +322,42 @@ CIGARETTE PACKETS ARE IN FANCY.DM
 		return
 	var/to_smoke = smoke_all ? (reagents.total_volume * (dragtime / smoketime)) : REAGENTS_METABOLISM
 	var/mob/living/carbon/smoker = loc
-	if(!istype(smoker) || src != smoker.wear_mask)
-		reagents.remove_any(to_smoke)
-		return
+	// These checks are a bit messy but at least they're fairly readable
+	// Check if the smoker is a carbon mob, since it needs to have wear_mask
+	if(!istype(smoker))
+		// If not, check if it's a gas mask
+		if(!istype(smoker, /obj/item/clothing/mask/gas))
+			reagents.remove_any(to_smoke)
+			return
+
+		var/atom/potential_smoker = smoker.loc
+
+		// If it is, check if that mask is on a carbon mob
+		if(!iscarbon(potential_smoker))
+			reagents.remove_any(to_smoke)
+			return
+
+		// If it is, check if that mob is actually wearing the mask
+		var/mob/living/carbon/mask_holder = potential_smoker
+		if(mask_holder.wear_mask != smoker)
+			reagents.remove_any(to_smoke)
+			return
+
+		// Update smoker to the wearer of the mask
+		smoker = potential_smoker
+
+	// Check if the smoker is actually wearing the cig
+	if(src != smoker.wear_mask)
+		// If they're not, check if the mask is a gas mask
+		if(istype(smoker.wear_mask, /obj/item/clothing/mask/gas))
+			// If it is, check if the cig is being held by that gas mask
+			var/obj/item/clothing/mask/gas/mask = smoker.wear_mask
+			if(src != mask.cig)
+				reagents.remove_any(to_smoke)
+				return
+		else
+			reagents.remove_any(to_smoke)
+			return
```

Comment:
This is so confounding, can't something like this be done instead? (I suggest testing the changes locally if necessary cause I cannot deny being a bit bewildered) : 
```suggestion
		smoker = smoker.loc

		// If it is, check if that mask is on a carbon mob
		if(!istype(smoker) || smoker.get_item_by_slot(ITEM_SLOT_MASK) != loc)
			reagents.remove_any(to_smoke)
			return
```
The rest is mostly fine though.

Reply:
Yeah that works, thanks